src/NServiceBus.Core/Saga/UniqueAttribute_obsolete.cs

```diff
@@ -17,19 +18,10 @@ public sealed class UniqueAttribute : Attribute
         /// </summary>
         /// <param name="type">The <see cref="Type"/> to evaluate.</param>
         /// <returns>A <see cref="PropertyInfo"/> of the property marked with a <see cref="UniqueAttribute"/> or null if not used.</returns>
+        [ObsoleteEx(RemoveInVersion = "7.0", TreatAsErrorFromVersion = "6.0", Message = "Use the new SagaMetadata")]
```

Comment:
Thanks for clarifying @simoncropp

On Mon, Apr 13, 2015 at 1:47 PM, Simon Cropp <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Saga/UniqueAttribute_obsolete.cs
> <https://github.com/Particular/NServiceBus/pull/2693#discussion_r28231556>
> :
>
> > @@ -17,19 +18,10 @@ public sealed class UniqueAttribute : Attribute
> >          /// </summary>
> >          /// <param name="type">The <see cref="Type"/> to evaluate.</param>
> >          /// <returns>A <see cref="PropertyInfo"/> of the property marked with a <see cref="UniqueAttribute"/> or null if not used.</returns>
> > +        [ObsoleteEx(RemoveInVersion = "7.0", TreatAsErrorFromVersion = "6.0", Message = "Use the new SagaMetadata")]
>
> should be enough to obsolete the class. unless the methods/props have more
> explicit alternatives and u want to show a different compile error
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2693/files#r28231556>.
>